sdv/_utils.py

```diff
@@ -81,6 +81,7 @@ def _is_datetime_type(value):
             bool(_get_datetime_format([value]))
```

Comment:
why doesn't guessing the datetime format for this string format work?

Reply:
It can't guess times.

Reply:
What do you mean by times?

Reply:
It's able to guess the following
```python
a = np.array(['09-12-2020 00:00:00'], dtype=object)
_guess_datetime_format_for_array(a.astype(object))
```
yields
```
'%m-%d-%Y %H:%M:%S'
```

Reply:
It also successfully guesses
```python
 a = np.array(['09-12-2020 00:00:00.010122'], dtype=object)
_guess_datetime_format_for_array(a.astype(object))
```
yields
```
'%m-%d-%Y %H:%M:%S.%f'
```

Reply:
Curious. `'2011-10-15 21:11:03.498707'` and `'2011-10-15 19:11:03.498707'` work, but it fails for `'2011-10-15 20:11:03.498707'`.

Reply:
@amontanez24 The issue happens when year and hour:minute have the same number (e.g., 2023 and 20:23). Pandas [addressed this bug 5 months ago](https://github.com/pandas-dev/pandas/pull/57471), but I don't know if they have released it yet (they are in the process of making a release [right now](https://github.com/pandas-dev/pandas/pull/59840), so it's possible this will get fixed in 2.2.3).

Reply:
Ah makes sense. I guess my follow up question is why use this function at all instead of just `pd.notna(pd.to_datetime(value, errors='coerce')))`? That seems to work for a lot of formats

Reply:
We can delete it. I left it to preserve previous behavior - I'm not certain `to_datetime` includes every case that `_get_datetime_format` does - but since it does pass every test case, we can probably delete it.